### ISSUE TITLE
Enable "pass through" entity fields

### ISSUE THREAD

**Issue by SHeimlich** [PolymerLabs/arcs#4561]
**The Goal**
I have a 3 particles that interact with an entity that has three fields. Let's call them output, process, and display. I want their interactions with the entity to be defined as follows:
```
output: writes Entity{a, b}
process: reads Entity{a}, writes Entity{a, c}
display: reads Entity{a, b, c}  
```
In other words, I want the output handle to write the a and b fields. Then the process particle should read a and write a and c. Finally, the display particle should read the entity which has a, b, and c. The a and c fields should be set by process, and b should be set by output.

In other words, the b field should pass through process, without process seeing it.

We tried doing this using type variable today, and it didn't seem to work. See the branch
[here:](https://github.com/PolymerLabs/arcs/compare/master...SHeimlich:type-variables?expand=1)
